src/Glpi/Search/Input/QueryBuilder.php

```diff
@@ -845,9 +848,66 @@ public static function cleanParams(array $params): array
             }
         }
 
+        self::validateCriteria($params);
+
         return $params;
     }
 
+    private static function validateCriteria(array &$params): void
+    {
+        if (!isset($params['criteria'])) {
+            return;
+        }
+        $valid_main_opts = SearchOption::getOptionsForItemtype($params['itemtype']);
+
+        // Validate criteria
+        $invalid_criteria = [];
+        foreach ($params['criteria'] as $k => $criterion) {
+            if (!array_key_exists('field', $criterion) || !is_numeric($criterion['field'])) {
+                continue;
+            }
+            if (isset($criterion['itemtype'])) {
+                // In the criteria array, the search options are from the metatype POV (Agent Name for example is ID 1 in criteria array, but 900 from the POV of Computer)
+                $valid_meta_opts = SearchOption::getOptionsForItemtype($criterion['itemtype']);
+                if (!isset($valid_meta_opts[(int) $criterion['field']])) {
+                    $invalid_criteria[] = $k;
+                    unset($params['criteria'][$k]);
+                }
+            } else if (!isset($valid_main_opts[(int) $criterion['field']])) {
+                $invalid_criteria[] = $k;
+                unset($params['criteria'][$k]);
+            }
+        }
+
+        // Validate sorts
+        if (isset($params['sort'])) {
+            if (!is_array($params['sort'])) {
+                $params['sort'] = [(int) $params['sort']];
+                $params['order'] = [$params['order'] ?? 'ASC'];
+            }
+            foreach ($params['sort'] as $k => $sorted_id) {
+                // Validate sort (IDs are always from the POV of the main itemtype)
+                if (!isset($valid_main_opts[$sorted_id])) {
+                    unset($params['sort'][$k], $params['order'][$k]);
+                }
+            }
+        }
+        if (empty($params['sort'])) {
+            $params['sort'] = [0];
+            $params['order'] = ['ASC'];
+        }
+
+        if (count($invalid_criteria) > 0) {
+            // There is probably no need to show more information about the invalid criteria
+            Session::addMessageAfterRedirect(__s('Some search criteria were removed because they are invalid'), false, WARNING);
+            if (GLPI_ENVIRONMENT_TYPE === GLPI::ENV_DEVELOPMENT) {
```

Comment:
IMHO, having a warning just for the developers is useless. The conditions are `a criterion refers to an invalid search option (typo in API request, missing plugin, bad saved search, etc)`, and I think that this kind of issue is more likely to happen on a production/staging GLPI instance.

This warning should probably be also triggerred when the debug mode is active, so the debug mode could be used even on a production env to understand what is the exact issue.

```suggestion
            if (
                GLPI_ENVIRONMENT_TYPE === GLPI::ENV_DEVELOPMENT
                || $_SESSION['glpi_use_mode'] === Session::DEBUG_MODE
            ) {
```